src/Dictionaries/MySQLDictionarySource.cpp

```diff
@@ -72,7 +74,11 @@ MySQLDictionarySource::MySQLDictionarySource(
     , query_builder{dict_struct, db, "", table, where, IdentifierQuotingStyle::Backticks}
     , load_all_query{query_builder.composeLoadAllQuery()}
     , invalidate_query{config.getString(config_prefix + ".invalidate_query", "")}
-    , close_connection{config.getBool(config_prefix + ".close_connection", false) || config.getBool(config_prefix + ".share_connection", false)}
+    , close_connection{
+            config.getBool(config_prefix + ".close_connection", false) || config.getBool(config_prefix + ".share_connection", false)}
+    , max_tries_for_mysql_block_input_stream{
+            config.getBool(config_prefix + ".fail_on_connection_loss", false)
+                    ? 1 : default_num_tries_on_connection_loss}
```

Comment:
maybe also add param `num_tries_on_mysql_connection_loss` to config ...

Reply:
I doubt that such parameter would actually be meaningful to users. The value of 3 is the common default for retries count. And thus only `.fail_on_connection_loss` is exposed for configuring.